api/filter_dimensions_test.go

```diff
@@ -203,6 +204,11 @@ func TestSuccessfulAddFilterBlueprintDimension(t *testing.T) {
 		Convey("Then the auditor is called for the attempt and outcome", func() {
 			assertAuditCalled(mockAuditor, addDimensionAction, actionSuccessful, expectedAuditParams)
 		})
+
+		Convey("Then the request body has been drained", func() {
+			_, err = r.Body.Read(make([]byte, 1))
```

Comment:
Possibly check body is nil instead of relying on there being an error, if there is no error you may want to know whether the body is nil or not as this will give you more to work with if it breaks.

Reply:
I have updated the test to also check the first returned value - the number of bytes read.